sys/random/prng_tinymt32.c

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+ /**
+ * @ingroup sys_random
+ * @{
+ * @file
+ *
+ * @brief Glue-code for tinymt32
+ *
+ * See https://github.com/MersenneTwister-Lab/TinyMT for details.
+ *
+ * @author Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "tinymt32/tinymt32.h"
+
+static tinymt32_t _random;
+
+void genrand_init(uint32_t seed)
+{
+    tinymt32_init(&_random, seed);
+}
```

Comment:
Just out of curiosity: have you checked if something like
```C
#define genrand_init(seed)    tinymt32_init(&_random, seed)
```
produces the same binary for various platforms?